app/javascript/components/patient/vaccines/VaccineTable.js

```diff
@@ -0,0 +1,421 @@
+  };
+
+  /**
+   * Makes a request to create a new vaccine on the backend and reloads page once complete.
+   * @param {*} newVaccineData
+   */
+  addNewVaccine = newVaccineData => {
+    axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+    axios
+      .post('/vaccines', {
+        group_name: newVaccineData.group_name,
+        product_name: newVaccineData.product_name,
+        administration_date: newVaccineData.administration_date,
+        dose_number: newVaccineData.dose_number,
+        notes: newVaccineData.notes,
+        patient_id: this.props.patient.id,
+      })
+      .then(() => {
+        // Refresh the page to see the updated table data
+        location.reload();
+      })
+      .catch(err => {
+        reportError(err?.response?.data?.error ? err.response.data.error : err, false);
+      });
+  };
+
+  /**
+   * Called when the edit vaccine button is clicked.
+   * Updates the state to show the appropriate modal for editing a vaccine.
+   */
+  handleEditVaccineClick = row => {
+    this.setState({
+      showEditVaccineModal: true,
+      editRow: row,
+    });
+  };
+
+  /**
+   * Closes the edit vaccine modal by updating state.
+   */
+  handleEditVaccineModalClose = () => {
+    this.setState({
+      showEditVaccineModal: false,
+      editRow: null,
+    });
+  };
+
+  /**
+   * Closes the EditVaccine modal and makes a request to update an existing vaccine record.
+   * @param {*} updatedVaccineData - State from vaccine modal containing updated vaccine data.
+   */
+  handleEditVaccineModalSave = updatedVaccineData => {
+    const currVaccineId = this.state.table.rowData[this.state.editRow]?.id;
+
+    this.setState(
+      {
+        showEditVaccineModal: false,
+        editRow: null,
+      },
+      () => {
+        this.editVaccine(updatedVaccineData, currVaccineId);
+      }
+    );
+  };
+
+  /**
+   * Makes a request to update an existing vaccine record on the backend and reloads page once complete.
+   * @param {*} newVaccineData
+   */
+  editVaccine = (updatedVaccineData, currVaccineId) => {
+    axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+
+    axios
+      .put('/vaccines/' + currVaccineId, {
+        group_name: updatedVaccineData.group_name,
+        product_name: updatedVaccineData.product_name,
+        administration_date: updatedVaccineData.administration_date,
+        dose_number: updatedVaccineData.dose_number,
+        notes: updatedVaccineData.notes,
+        patient_id: this.props.patient.id,
+      })
+      .then(() => {
+        // Refresh the page to see the updated table data
+        location.reload();
+      })
+      .catch(err => {
+        reportError(err?.response?.data?.error ? err.response.data.error : err, false);
+      });
+  };
+
+  /**
+   * Formats values in the timestamp column to be human readable
+   * @param {Object} data - Data about the cell this filter is called on.
+   */
+  formatTimestamp(data) {
```

Comment:
It's probably worth centralizing this as a helper somewhere.

Reply:
I've honestly been dying to do this for awhile. I actually made a [ticket](https://tracker.codev.mitre.org/browse/SARAALERT-654) for it on the backlog in August (lol), just haven't had a chance to actually get a PR up for it. The reason I didn't do it here is because this PR is already quite large and I assumed I'd want to pull this function out from all the other files it was in as well. If we're fine with the additional file changes, I'm okay with adding it and getting it over and done with - or I can just pull this backlog ticket into next sprint.

Reply:
Update: @DPC15038  will be working on this ticket so whatever work that goes in first can handle this change.